### PR TITLE
update kustomize scaffolding to work with kustomize 2.0

### LINKED ISSUE #595: make deploy is broken for kustomize v2.0.0 release
kustomize v2.0.0 requires that all paths other than bases have to terminate in same or subdirectory of the kustomization directory. So `kustomize build config/default` in `make deploy` fails because `kustomization.yaml` contains resource paths like `../rbac/rbac_role.yaml`. According to kustomize v2.0.0, `rbac` and `manager` folder should be under `config/default` directory.

### PR DESCRIPTION
**Pull request by Liujingfang1**
Fix #595

### CODE REVIEW

**Comment by droot** [comment]
@Liujingfang1 Will this new structure work with kustomize 1.0.0 as well ? 

I am thinking of cases where users have projects with current/new kustomization structure, but they have older kustomize (1.0.0) will it work for both ? because upgrading to kustomize 2.0.0 will force them to migrate to newer structure.

Also, was wondering if we can update the issue with set of kustomize commands to run to migrate the old structure to work with new structure.

**Comment by Liujingfang1** [comment]
@droot Yes, this structure works with kustomize 1.0 as well.

**Comment by Liujingfang1** [comment]
> kustomize commands to run to migrate the old structure to work with new structure

There is no commands in Kustomize to migrate the structures. It requires manual work :(

**Comment by Liujingfang1** [comment]
@droot  I added some instructions for migrating from the old structure to the new structure in the issue.

File: pkg/scaffold/project/kustomize.go [hunk 1]
```diff
@@ -65,21 +65,9 @@ namePrefix: {{.Prefix}}-
...
-- ../rbac/rbac_role_binding.yaml
-- ../manager/manager.yaml
-  # Comment the following 3 lines if you want to disable
-  # the auth proxy (https://github.com/brancz/kube-rbac-proxy)
-  # which protects your /metrics endpoint.
-- ../rbac/auth_proxy_service.yaml
-- ../rbac/auth_proxy_role.yaml
-- ../rbac/auth_proxy_role_binding.yaml
+bases:
+- ../rbac
```
Comment (droot) [inline pkg/scaffold/project/kustomize.go]: I have to double check, but rbac manifests get generated everytime make runs, so not sure if delete all yamls in `rbac` directory.

Comment (mengqiy) [reply pkg/scaffold/project/kustomize.go]: IIRC it doesn't delete everything, it overwrites the content in `../rbac/rbac_role.yaml`. We'd better confirm it.

Comment (droot) [reply pkg/scaffold/project/kustomize.go]: I confirmed, it doesn't delete, so the change is good.

**Comment by droot** [comment]
@Liujingfang1 The change is good, but we do not have an e2e test in travis which can verify this. we generally test `kustomize` related changes manually. Will you be able to perform a manual test before the merge ?

**Comment by Liujingfang1** [comment]
@droot Sure, I'll test it manually :)

**Comment by Liujingfang1** [comment]
I ran following commands and all of them succeeded.
```
kubebuilder init --domain jingfang
kubebuilder create api --group my --version v1 --kind MyKind --resource=true --controller=true
make install
make run
kubectl apply -f config/samples/.

# deploy the manager to cluster
make docker-build IMG=jiliu/kb
make docker-push IMG=jiliu/kb
make deploy
```

**Comment by droot** [comment]
Awesome. Thanks for the e2e test.

**Pull request merged**